src/runner.ts

```diff
@@ -173,19 +173,28 @@ export class CoverageParserRunner {
     }
 
     private async isCoverageReport(report: string): Promise<boolean> {
-        const X2jOptions = {
-            ignoreAttributes: false,
-            parseAttributeValue: true
-        }
-
-        const xmlData = fs.readFileSync(report);
-        const parser = new XMLParser(X2jOptions);
-        const parsedXml = parser.parse(xmlData);
-
-        if (parsedXml.Coverage && parsedXml.Coverage['@_ver']) {
-            return true;
-        }
-        return false;
+        return new Promise((resolve) => {
+            let isCoverageReport = false;
+            const saxStream = sax.createStream(true, {});
+            saxStream.on("opentag", (node) => {
+                if (!isCoverageReport && node.name == 'Coverage' && node.attributes.hasOwnProperty('ver')) {
+                    core.debug(messagesFormatter.format(messages.recognized_coverage_report, report));
+                    isCoverageReport = true;
```

Comment:
Is it better to stop the stream here since we've already found the Coverage tag?
Consider to call `saxStream.close()` here.

Reply:
The close function is not exist in saxStream. We can use `saxStream.end()` to close the stream. But the stream cannot be closed immediately by using `saxStream.end()`, it continues to process the XML data that has already been buffered and triggers the remaining events. Also, some errors may be caused after calling `saxStream.end()`.
![warning_message](https://github.com/user-attachments/assets/3fbb3388-2eee-4d07-92eb-d0361f7d91ce)

Reply:
I see, you are right. I was wondering if there's a way to close the stream earlier, but after searching on Google, it seems there isn't a good way to do it, and we are not supposed to do it.